test.js

```diff
@@ -1,8 +1,359 @@
-import test from 'ava';
-import m from './';
+import {serial as test} from 'ava';
+import mem from 'mem';
+import mock from 'mock-require';
+
+let originalEnv;
+let originalPlatform;
+test.before(() => {
+	originalEnv = Object.assign({}, process.env);
+	originalPlatform = Object.getOwnPropertyDescriptor(process, 'platform');
+});
+
+test.beforeEach(() => {
+	delete process.env.GIT_AUTHOR_NAME;
+	delete process.env.GIT_COMMITTER_NAME;
+	delete process.env.HGUSER;
+	delete process.env.C9_USER;
+
+	Object.defineProperty(process, 'platform', originalPlatform);
+	mock('rc', () => ({}));
+});
+
+test.after(() => {
+	process.env = Object.assign({}, originalEnv);
+	Object.defineProperty(process, 'platform', originalPlatform);
+});
+
+test('should get fullname from env var', async t => {
+	const m = require('./');
+	mem.clear(m);
+
+	process.env.GIT_AUTHOR_NAME = 'TEST-ENV-FULL-NAME';
+	const fullname = await m();
+
+	t.is(typeof fullname, 'string');
+	t.is(fullname, 'TEST-ENV-FULL-NAME');
+});
+
+test('should get value from init-author-name', async t => {
+	mock('rc', () => ({
+		'init-author-name': 'TEST-INIT-AUTHOR-FULL-NAME'
+	}));
+
+	const m = require('./');
+	mem.clear(m);
+
+	const fullname = await m();
+
+	t.is(typeof fullname, 'string');
+	t.is(fullname, 'TEST-INIT-AUTHOR-FULL-NAME');
+});
+
+test('should get value from passwdUser for darwin platform', async t => {
+	// redefine process.platform
+	Object.defineProperty(process, 'platform', {
+		value: 'darwin'
+	});
+
+	mock('passwd-user', () => Promise.resolve({
+		fullname: 'TEST-PASSWD-FULL-NAME'
+	}));
+
+	// Required to delete require cache so the mock to passwd-usr is taken into consideration
+	delete require.cache[require.resolve('./')];
```

Comment:
Use https://github.com/sindresorhus/require-uncached

Same with the other places this is done.